yagm.go

```diff
@@ -23,8 +25,9 @@ type yagmRequest struct {
 // against a list of registered patterns and calls
 // the handler for the pattern that matches the URL.
 type YagmMux struct {
-	mu     sync.RWMutex
-	routes map[string]*yagmRoute
+    mu     sync.RWMutex
+    routes map[string]*yagmRoute
+    cache  map[hash.Hash]string
```

Comment:
You can speed up things with a `*yagmRoute` cache, You can save one extra access  to `routes`.